.github/workflows/ci-tests.yml

```diff
@@ -0,0 +1,137 @@
+name: Go Test Workflow
+
+on:
+  push:
+    branches: [master]
+  pull_request:
+    branches: [master]
+env:
+  TYK_IB_STORAGE_STORAGETYPE: file
+  
+permissions:
+  contents: read
+  
+jobs:
+  golangci-lint:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Tyk Identity Broker
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ github.event.pull_request.head.ref }}
+      - name: Fetch base branch
+        if: ${{ github.event_name == 'pull_request' }}
+        run: git fetch origin ${{ github.base_ref }}
+      - name: golangci-lint
+        if: ${{ github.event_name == 'pull_request' }}
+        uses: golangci/golangci-lint-action@v3
+        with:
+          version: latest
+          args: --out-format=checkstyle:golanglint.xml --timeout=600s --max-issues-per-linter=0 --max-same-issues=0 --new-from-rev=origin/${{ github.base_ref }}
+      - uses: actions/upload-artifact@v2
+        with:
+          name: golangcilint
+          retention-days: 1
+          path: |
+            golanglint.xml
+  
+  ci-test:
+    name: '${{ matrix.databases }}'
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        databases:
+          - mongo-mgo
+          - mongo-official
+          - file
+        redis-version: [5]
+        mongodb-version: [4.2]
+    steps:
+     - name: Setup Go
+       uses: actions/setup-go@v2
+       with:
+          go-version: '^1.19'
+    
+     - name: Checkout Code
+       uses: actions/checkout@v4
+       with:
+        ref: ${{ github.event.pull_request.head.ref }}
+
+     - name: Fetch base branch
+       if: ${{ github.event_name == 'pull_request' }}
+       run: git fetch origin ${{ github.base_ref }}
+
+     - name: Install Dependencies and basic hygiene test
+       id: hygiene
+       run: |
+        go install golang.org/x/tools/cmd/goimports@latest
+     
+     - name: Start Redis
+       uses: supercharge/redis-github-action@1.2.0
+       with:
+          redis-version: ${{ matrix.redis-version }}
+
+     - name: Start MongoDB
+       uses: supercharge/mongodb-github-action@1.2.0
+       with:
+          mongodb-version: '${{ matrix.mongodb-version }}'
+          
+     - name: Run tests
+       run: |
+        ./bin/ci-tests.sh ${{ matrix.databases }}
+     - uses: actions/upload-artifact@v2
+       with:
+         name: coverage
+         retention-days: 1
+         path: |
+           *cov
+  
+  sonar-cloud-analysis:
```

Comment:
Suggest merging this job with the golangci-lint job which will save you the trouble of transferring artefacts via {upload,download}-artifact. It can depend on ci-test if you want.